Assets/Mirror/Runtime/NetworkManagerHUD.cs

```diff
@@ -154,5 +228,60 @@ void StopButtons()
                 }
             }
         }
+
+        void SetIPInformation()
+        {
+            if (_IPType == IPType.localhost)
+            {
+                inputField = "localhost";
+                manager.networkAddress = inputField;
+            }
+            else if (_IPType == IPType.LAN)
+            {
+                StartCoroutine(GetLanAddress());
+            }
+            else if (_IPType == IPType.WAN)
+            {
+                StartCoroutine(GetWanAddress());
+            }
+        }
+
+        IEnumerator GetLanAddress()
+        {
+            if (lanIP != "") { inputField = lanIP; yield break; }
+            
+            #if UNITY_IOS
+            lanIP = System.Net.NetworkInformation.NetworkInterface.GetAllNetworkInterfaces().Where(x => x.Name.Equals("en0")).First().GetIPProperties().UnicastAddresses.Where(x => x.Address.AddressFamily == AddressFamily.InterNetwork).First().Address.ToString();
+            #else
+            lanIP = Dns.GetHostEntry(Dns.GetHostName()).AddressList.FirstOrDefault(ip => ip.AddressFamily == AddressFamily.InterNetwork).ToString();
+            #endif
+            
+            if (lanIP != "")
+            {
+                inputField = lanIP;
+                manager.networkAddress = lanIP;
+            }
+        }
+        
+        IEnumerator GetWanAddress()
```

Comment:
same here. please avoid coroutine

Reply:
I think this has to be a coroutine because it uses UnityWebRequest that won't block